front50-gcs/src/main/java/com/netflix/spinnaker/front50/model/GcsStorageService.java

```diff
@@ -228,27 +231,40 @@ public GcsStorageService(String bucketName,
     this.registry = registry;
 
     Id id = registry.createId("google.storage.invocation");
-    deleteTimer = registry.timer(id.withTag("method", "delete"));
-    purgeTimer = registry.timer(id.withTag("method", "purgeTimestamp"));
-    loadTimer = registry.timer(id.withTag("method", "load"));
-    listTimer = registry.timer(id.withTag("method", "list"));
-    mediaDownloadTimer = registry.timer(id.withTag("method", "mediaDownload"));
-    insertTimer = registry.timer(id.withTag("method", "insert"));
-    patchTimer = registry.timer(id.withTag("method", "patch"));
+    deleteTimer = id.withTag("method", "delete");
+    purgeTimer = id.withTag("method", "purgeTimestamp");
+    loadTimer = id.withTag("method", "load");
+    listTimer = id.withTag("method", "list");
+    mediaDownloadTimer = id.withTag("method", "mediaDownload");
+    insertTimer = id.withTag("method", "insert");
+    patchTimer = id.withTag("method", "patch");
   }
 
-  private <T> T timeExecute(Timer timer, AbstractGoogleClientRequest<T> request) throws IOException {
+  private <T> T timeExecute(Id timerId, AbstractGoogleClientRequest<T> request) throws IOException {
+     T result;
+     Clock clock = registry.clock();
+     long startTime = clock.monotonicTime();
+     int statusCode = -1;
+
      try {
-       return timer.record(new Callable<T>() {
-          public T call() throws IOException {
-            return request.execute();
-          }
-       });
+       result = request.execute();
+       statusCode = request.getLastStatusCode();
+     } catch (HttpResponseException e) {
+       statusCode = e.getStatusCode();
+       throw e;
      } catch (IOException ioex) {
-         throw ioex;
+       throw ioex;
      } catch (Exception ex) {
-         throw new IllegalStateException(ex);
+       throw new IllegalStateException(ex);
+     } finally {
+       long nanos = clock.monotonicTime() - startTime;
+       String status = Integer.toString(statusCode).charAt(0) + "xx";
```

Comment:
In the `deserialize` function, we replace `status` with `200` if it's `<0` (ie, no request was made).  Should we do the same here?

Reply:
The 200 in deserialize is working around in a bug where  a request *is* made and -1 is being returned by the object for some reason. I threw a 200 in there for speculation so it's easier to deal with (I was going to add something like 299 so it would be 2xx but clearly weird but decided against that). The default -1 value would indicate an exception thrown for reasons other than a bad HttpStatus.